src/writers.ts

```diff
@@ -8,15 +14,28 @@ export function createEvmWriters(indexerName: string) {
 
     const sender = event.args.sender;
     const token = event.args.token;
-    const amount = event.args.amount;
+    const amount = (Number(event.args.amount.toString()) / 1e6).toFixed(2); // Transforms 1990000 to 1.99
     const barcode = event.args.barcode;
 
-    const payment = new Payment(`${sender}/${tx.hash}`, indexerName);
+    const chain = indexerName; // The indexer name corresponds to the chain name
+    let tokenName = getTokenName(token, chain);
+
+    if (!tokenName) {
+      // TODO log missing token
+      tokenName = token;
+    }
+
+    const payment = new Payment(`${tx.hash}`, indexerName);
     payment.sender = sender;
-    payment.token = token;
+    payment.token = tokenName;
     payment.amount = amount;
```

Comment:
Can we store both raw and decimal amount?

Reply:
1a3e6e7852ad08dc88e2c135a786e2a40c4b8da3